packages/fullstack/docs/request_origin.md

```diff
@@ -0,0 +1,43 @@
+This method interacts with information from the current request. The request may come from:
+
+1. The initial SSR render if this method called from a [`Component`](dioxus_lib::prelude::component) or a [`server`](crate::prelude::server) function that is called during the initial render
+
+```rust
+#[component]
+fn PrintHtmlRequestInfo() -> Element {
+    // The server context only exists on the server, so we need to put it behind a server_only! config
+    server_only! {
+        // Since we are calling this from a component, the server context that is returned will be from
+        // the html request for ssr rendering
+        println!("headers are {:?}", server_context().request_parts().headers);
+    }
+    rsx! {}
+}
+```
+
+2. A request to a [`server`](crate::prelude::server) function called directly from the client (either on desktop/mobile or on the web frontend after the initial render)
+
+```rust
+#[server]
+async fn read_headers() -> Result<(), ServerFnError> {
+    // The server context only exists on the server, so we need to put it behind a server_only! config
+    server_only! {
```

Comment:
Is `server_only!` redundant already being in a `#[server]` defined server function?